src/browser.ts

```diff
@@ -0,0 +1,148 @@
+import puppeteer, { ElementHandle } from 'puppeteer';
+import { Page, Browser as PuppeteerBrowser } from 'puppeteer';
+import { sleep } from './util/sleep';
+import { LikeOptions } from './luckybot';
+
+const host = 'https://www.instagram.com';
+
+export interface BrowserOptions {
+  debug?: boolean;
+}
+
+export class Browser {
+  browser?: PuppeteerBrowser;
+  page?: Page;
+  isLoggedIn: boolean = false;
+  options: BrowserOptions;
+
+  constructor(options?: BrowserOptions) {
+    this.options = options || {debug: false};
+  }
+
+  async createPage() {
+    const browser = await puppeteer.launch({ headless: !this.options.debug });
+    const page = await browser.newPage();
+    const userAgent = await browser.userAgent();
+    await page.setViewport({width: 1024, height: 768});
+    await page.setUserAgent(userAgent.replace('Headless', ''));
+    
+    this.browser = browser;
+    this.page = page;
+
+    return page;
+  }
+
+  getPage(): Promise<Page> {
+    const {page} = this;
+
+    if (page) return Promise.resolve(page);
+
+    return this.createPage();
+  }
+
+  async login(userName: string, password: string) {
+    if (this.isLoggedIn) return;
+
+    const page = await this.getPage();
+
+    await page.goto(`${host}/accounts/login`);
+
+    const userInput = await page.waitForSelector('input._2hvTZ');
+    const passInput = await page.waitForSelector('input[type=password]');
+    const logInButton = await page.waitForSelector('button[type="submit"]');
+
+    if (!userInput || !passInput || !logInButton) {
+      throw new Error('Error loging');
+    }
+
+    await userInput.type(userName);
+    await passInput.type(password);
+    await logInButton.click()
+    await page.waitForNavigation();
+
+    this.isLoggedIn = true;
+  }
+
+  async getPhotos(hashtag: string) {
+    const page = await this.getPage();
+
+    await page.goto(`${host}/explore/tags/${hashtag}/`);
```

Comment:
I tought other than login all the other stuff would use private APIs no?

Reply:
private you mean within this class? Or to move that somewhere else? 

Currently I didn't add any private, let me fix that

Reply:
No I mean: here we use scraping for all, but I tought we were going to use private APIs except for login

Reply:
In strategy 1 (Scraper) we are using scraping for all thats correct